src/ua/artcode/udiary/dao/UserDaoImpl.java

```diff
@@ -23,26 +25,36 @@ public UserDaoImpl(AppDataContainer appDataContainer) {
 
     @Override
     public User save(User user) {
-        return null;
+
+        Long generatedKey = UUID.randomUUID().getLeastSignificantBits();
+
+        user.setId(generatedKey);
+        appDataContainer.userMap.put(generatedKey, user);
+
+        return user;
     }
 
     @Override
     public User findOne(Long aLong) {
-        return null;
+        return appDataContainer.userMap.get(aLong);
     }
 
     @Override
     public List<User> findAll() {
-        return null;
+        return new ArrayList<>(appDataContainer.userMap.values());
     }
 
     @Override
     public User delete(Long aLong) {
-        return null;
+        return appDataContainer.userMap.remove(aLong);
     }
 
     @Override
     public User update(User user) {
-        return null;
+        return appDataContainer.userMap.entrySet().stream()
+                .filter(el ->
+                        el.getValue().getId() == user.getId()
+                )
+                .findFirst().get().setValue(user);
```

Comment:
Why do we set a user via setValue? Is that business case?

Reply:
My bad. I've fixed it. You can check the new version in the next push.